### PR TITLE
LiveVis: Add Server

### PR DESCRIPTION
**Pull request by ax3l**
This is the first tool of our 3D live vis release, the server.
It can be compiled with the recent release of [rivlib](https://github.com/ComputationalRadiationPhysics/rivlib).

# State of this branch

@psychocoderHPC and me cleaned, fixed and upgraded this tool.

It uses a very simple `Makefile` and a hard-coded `g++` so far but should be extremely easy to compile.
A `README.md` file was added, please refer to it.

# Acknowledgments

A big thank you to @Ben-Schneider as the author of the tool!
Kudos also to @SGrottel-TUD and  as the author of [rivlib](https://github.com/ComputationalRadiationPhysics/rivlib) and to @tud-mkalms for the support and new image stream implementations last year.

### CODE REVIEW

**Comment by ax3l** [comment]
A note on the tests.
I added `rivlib 1.0.0` to the compile suite with version but our `tools/` are not tested right now anyway.

We might add auto-tests with travis-ci after #586 or with #466, whatever comes first.

File: src/tools/livevis/server/include/net/message_ids.hpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    TimeStep,                   // the message contains one uint telling us the current time step
+    VisName,                    // the message contains a string (chars) giving the name of the simulation
+    CameraPosition,             // three floats determining the camera position
+    CameraOrbit,                // three floats determining a camera movement vector
+    CameraPan,                  // three floats determining orbiting the focal point around the camera position
+    CameraSlide,                // three floats moving the camera and focal point at the same time
+    CameraFocalPoint,           // three float determining the look at point of the camera
+    CameraDefault,              // message telling to reset the camera view to default
+    Weighting,                  // one float telling the weighting of the two data sources
+    SimPlay,					// run the simulation
```
Comment (PrometheusPi) [inline src/tools/livevis/server/include/net/message_ids.hpp]: tabs instead of spaces are used here

File: src/tools/livevis/server/main.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+{
+    int port = VISSERVER_DEFEULT_PORT;
+    int info_port = VISSERVER_DEFAULT_INFO_PORT;
+    
+    if (argc == 2)
+    {
+    	if ( strcmp(argv[1], "--help") == 0 || strcmp(argv[1], "-h") == 0 )
+    	{
+    		/// list help
+    		std::cout << "In Situ Visualization Server Help" << std::endl;
```
Comment (PrometheusPi) [inline src/tools/livevis/server/main.cpp]: tab instead of spaces is used here

File: src/tools/livevis/server/src/VisualizationServer.cpp [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+
+	//			::sendto(srv->m_info_socket, uri, 1024, 0, (struct sockaddr*)&remote_addr, sizeof(remote_addr));
+
+                std::cout << "Vis: " << namestr << " " << uristr << std::endl;
+		std::cout << "Combined: " << name_uri << std::endl;
+            }
+		}
+    }
+
+    /*while(1)
```
Comment (PrometheusPi) [inline src/tools/livevis/server/src/VisualizationServer.cpp]: why is this part of the code commented?

Comment (ax3l) [inline src/tools/livevis/server/src/VisualizationServer.cpp]: idk

**Comment by ax3l** [comment]
### to do

- [ ] remove dead code
- [ ] convert tabs to spaces ("author" this commit to Benjamin)

**Comment by ax3l** [comment]
@ComputationalRadiationPhysics/picongpu-maintainers pickme & merge me :)

**Comment by PrometheusPi** [comment]
I will have a look through the code again and will merge this later today.

**Comment by ax3l** [comment]
feel free to propose a pull request instead of annotations, since it's not rly my code anyway :)

**Comment by ax3l** [comment]
@PrometheusPi reminder :)

**Pull request merged**